### PR TITLE
Fixed abha linking issue

### LINKED ISSUE #8542: ABDM profile card and care context is missing in staging
**Describe the bug**

currently, the ABDM linking workflow in patient registration form is working, but all the other workflows from context to viewing Abha profile seem broken,

note: patient registration form , the ABDM address is being fetched, but post creation the linking is missing in the consultation

**To Reproduce**

**Expected behavior**

- [ ] all the ABDM workflow need to be working properly

**Screenshots**

image

### PR DESCRIPTION
**Pull request by khavinshankar**
## Proposed Changes

- Fixed 404 issue in get abha_number.
- Added link_patient api

### Associated Issue

- Fixes https://github.com/ohcnetwork/care_fe/issues/8542

@ohcnetwork/care-backend-maintainers @ohcnetwork/care-backend-admins

### CODE REVIEW

**Review by sainak — APPROVED** [review]

File: care/abdm/api/viewsets/abha_number.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def get_object(self):
             Q(abha_number=id) | Q(health_id=id) | Q(patient__external_id=id)
         ).first()
 
-        if not instance or get_patient_queryset(self.request.user).contains(
+        if not instance or not get_patient_queryset(self.request.user).contains(
```
Comment (anroopak) [inline care/abdm/api/viewsets/abha_number.py:31]: Can you add a relevant test case for this change?

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Comment by nihal467** [comment]
LGTM

File: care/abdm/api/viewsets/abha_number.py (hunk 1 above)
Comment (khavinshankar) [reply care/abdm/api/viewsets/abha_number.py:31]: Currently, we don't have any tests for ABDM, opened up an issue (https://github.com/ohcnetwork/care_abdm/issues/12) to track it.

**Review by anroopak — APPROVED** [review]

**Review by gigincg — APPROVED** [review]

**Pull request merged**